truffle/src/main/java/org/jruby/truffle/core/string/StringLiteral.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2016 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 1.0
+ * GNU General Public License version 2
+ * GNU Lesser General Public License version 2.1
+ */
+package org.jruby.truffle.core.string;
+
+import com.oracle.truffle.api.CompilerDirectives;
+import com.oracle.truffle.api.CompilerDirectives.CompilationFinal;
+import com.oracle.truffle.api.object.DynamicObject;
+import org.jcodings.specific.USASCIIEncoding;
+import org.jruby.truffle.RubyContext;
+import org.jruby.truffle.core.rope.CodeRange;
+import org.jruby.truffle.core.rope.Rope;
+
+import java.nio.charset.StandardCharsets;
+
+public class StringLiteral {
+
+    private final RubyContext context;
+    private final String literal;
+
+    @CompilationFinal private volatile Rope rope;
+
+    public StringLiteral(RubyContext context, String literal) {
+        assert is7Bit(literal);
+        this.context = context;
+        this.literal = literal;
+    }
+
+    public Rope getRope() {
+        if (rope == null) {
+            CompilerDirectives.transferToInterpreterAndInvalidate();
+            rope = createRope();
+        }
+
+        return rope;
+    }
+
+    public DynamicObject getString() {
```

Comment:
I think this method is a bit confusingly named. The wrapper object is a Java `String`, but this returns a Ruby `String`. Additionally, it's not a simple getter since it'll always create a new Ruby `String`.